### ISSUE TITLE
PS D:\wechaty\wechaty-getting-started> npm install安装失败

### ISSUE THREAD

**Issue by livingbody** [wechaty/wechaty-getting-started#196]
🚨 The issue tracker is not for questions 🚨

If you have a question, please ask it on https://stackoverflow.com/questions/tagged/wechaty

[question]
`9 verbose type range
2710 verbose stack wechaty-puppet-whatsapp: No matching version found for wechaty-puppet-whatsapp@^0.2.4.
2710 verbose stack     at module.exports (C:\Users\livingbody\AppData\Roaming\npm\node_modules\npm\node_modules\npm-pick-manifest\index.js:209:23)
2710 verbose stack     at C:\Users\livingbody\AppData\Roaming\npm\node_modules\npm\node_modules\pacote\lib\registry.js:118:26
2710 verbose stack     at async Arborist.[nodeFromEdge] (C:\Users\livingbody\AppData\Roaming\npm\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\build-ideal-tree.js:936:19)
2710 verbose stack     at async Arborist.[buildDepStep] (C:\Users\livingbody\AppData\Roaming\npm\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\build-ideal-tree.js:882:11)
2710 verbose stack     at async Arborist.buildIdealTree (C:\Users\livingbody\AppData\Roaming\npm\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\build-ideal-tree.js:218:7)
2710 verbose stack     at async Promise.all (index 1)
2710 verbose stack     at async Arborist.reify (C:\Users\livingbody\AppData\Roaming\npm\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\reify.js:134:5)
2710 verbose stack     at async Install.install (C:\Users\livingbody\AppData\Roaming\npm\node_modules\npm\lib\install.js:150:5)
2711 verbose cwd D:\wechaty\wechaty-getting-started
2712 verbose Windows_NT 10.0.19043
2713 verbose argv "D:\\Program Files\\nodejs\\node.exe" "C:\\Users\\livingbody\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "install"
2714 verbose node v16.4.2
2715 verbose npm  v7.19.1
2716 error code ETARGET
2717 error notarget No matching version found for wechaty-puppet-whatsapp@^0.2.4.
2718 error notarget In most cases you or one of your dependencies are requesting
2718 error notarget a package version that doesn't exist.
2719 verbose exit 1`

**Comment by zhaobingss** [comment]
同样遇到此问题

**Comment by goldcome** [comment]
同上，node版本v16.11.1，好难运行起来，太乱了，各种不兼容

**Comment by huan** [comment]
@goldcome @lixiyu @zhaobingss For easy getting started, I'd like to suggest that you can following our official getting start guide at https://wechaty.js.org/2021/02/06/wechaty-getting-started-without-leave-your-browser/

## TL;DR: [![Click me](https://img.shields.io/badge/Gitpod-Ready--to--Code-blue?logo=gitpod)](https://gitpod.io/#https://github.com/wechaty/wechaty-getting-started)

**Comment by lixiyu** [comment]
@huan 
I meet the problem at last week ,It also can‘t running in Gitpod something going wrong in nodejs environment .
but it runnable in this week

**Comment by huan** [comment]
@lixiyu Thanks for the reporting.

We are upgrading versions in the past week and there might be some version mismatch problem.

Please feel free to let me know if you have any problems in the current getting started repo, especially in the GitPod & google cloud shell environments because they are reproducible.

**Comment by lixiyu** [comment]
@huan
Ok, Thank You. I will be more free in next time

**Comment by huan** [comment]
You are welcome!

**Comment by goldcome** [comment]
我的问题是出现npm ERR! notarget No matching version found for wechaty-puppet@^1.2.7. 我用的是淘宝源，后面发现淘宝同步的版本只到1.0.1，我手动同步下淘宝源，然后现在没有问题了

**Comment by huan** [comment]
@goldcome Glad to know that it works for you now!
